kotlinx-coroutines-core/jvm/src/Future.kt

```diff
@@ -5,42 +5,22 @@ package kotlinx.coroutines
 
 import java.util.concurrent.*
 
-/**
- * Cancels a specified [future] when this job is cancelled.
- * This is a shortcut for the following code with slightly more efficient implementation (one fewer object created).
- * ```
- * invokeOnCompletion { if (it != null) future.cancel(false) }
- * ```
- *
- * @suppress **This an internal API and should not be used from general code.**
- */
-@InternalCoroutinesApi
-public fun Job.cancelFutureOnCompletion(future: Future<*>): DisposableHandle =
-    invokeOnCompletion(handler = CancelFutureOnCompletion(future))
-
 /**
  * Cancels a specified [future] when this job is cancelled.
  * This is a shortcut for the following code with slightly more efficient implementation (one fewer object created).
  * ```
  * invokeOnCancellation { if (it != null) future.cancel(false) }
  * ```
  */
+@Deprecated(
+    "This function does not do what its name implies: it will not cancel the future if just cancel() was called.",
```

Comment:
@zuevmaxim,
```suggestion
    "This function does not do what its name implies: it will not cancel the future if the continuation was cancelled without a cause specified (e.g., via `cancel()` with no argument).",
```
?

Reply:
Looks good

Reply:
It seems like this change wasn't actually applied?